libcudacxx/include/cuda/annotated_ptr

```diff
@@ -268,15 +268,14 @@ apply_access_property(const volatile void* __ptr, const _Shape __shape, access_p
     NV_PROVIDES_SM_80,
     (if (!__isGlobal((void*) __ptr)) return;
 
-     char* __p                               = reinterpret_cast<char*>(const_cast<void*>(__ptr));
-     static constexpr std::size_t _LINE_SIZE = 128;
-     std::size_t __nbytes                    = static_cast<std::size_t>(__shape);
-     std::size_t __end = ((std::uintptr_t)(__p + __nbytes) % _LINE_SIZE) ? __nbytes + _LINE_SIZE : __nbytes;
-     __end /= _LINE_SIZE;
+     uint8_t* __p                      = reinterpret_cast<uint8_t*>(const_cast<void*>(__ptr));
+     constexpr std::size_t __line_size = 128;
+     const std::size_t __nbytes        = static_cast<std::size_t>(__shape);
+     std::size_t __end                 = 1 + (__nbytes / __line_size);
```

Comment:
should this be `cuda::ceil_div(__nbytes, __line_size)`?

That Happens here if `__nbytes `is divisible by `__line_size`